Linker error FreeRTOSConfig.h with gcc 10

Linking examples fails because of [`modm_extern_c uint32_t _ZN4modm8platform4fcpuE;`](https://github.com/modm-io/modm/blob/develop/ext/aws/FreeRTOSConfig.h.in#L39) in FreeRTOSConfig.h
When compiling the header as C, `modm_extern_c` is defined as empty. Thus, we have a **definition** inside a header file. If we include this file in multiple C translation units, there will be multiple definitions and linking fails. That is the reason we see linker errors for freertos C files but not for C++ sources.

```
ld: build/release/modm/libmodm.a(list.o):/root/project/examples/stm32f746g_discovery/rtos/modm/ext/freertos/inc/freertos/FreeRTOSConfig.h:39: multiple definition of `modm::platform::fcpu'; build/release/modm/libmodm.a(croutine.o):/root/project/examples/stm32f746g_discovery/rtos/modm/ext/freertos/inc/freertos/FreeRTOSConfig.h:39: first defined here
ld: build/release/modm/libmodm.a(port.o):/root/project/examples/stm32f746g_discovery/rtos/modm/ext/freertos/inc/freertos/FreeRTOSConfig.h:39: multiple definition of `modm::platform::fcpu'; build/release/modm/libmodm.a(croutine.o):/root/project/examples/stm32f746g_discovery/rtos/modm/ext/freertos/inc/freertos/FreeRTOSConfig.h:39: first defined here
ld: build/release/modm/libmodm.a(queue.o):/root/project/examples/stm32f746g_discovery/rtos/modm/ext/freertos/inc/freertos/FreeRTOSConfig.h:39: multiple definition of `modm::platform::fcpu'; build/release/modm/libmodm.a(croutine.o):/root/project/examples/stm32f746g_discovery/rtos/modm/ext/freertos/inc/freertos/FreeRTOSConfig.h:39: first defined here
ld: build/release/modm/libmodm.a(stream_buffer.o):/root/project/examples/stm32f746g_discovery/rtos/modm/ext/freertos/inc/freertos/FreeRTOSConfig.h:39: multiple definition of `modm::platform::fcpu'; build/release/modm/libmodm.a(croutine.o):/root/project/examples/stm32f746g_discovery/rtos/modm/ext/freertos/inc/freertos/FreeRTOSConfig.h:39: first defined here
ld: build/release/modm/libmodm.a(tasks.o):/root/project/examples/stm32f746g_discovery/rtos/modm/ext/freertos/inc/freertos/FreeRTOSConfig.h:39: multiple definition of `modm::platform::fcpu'; build/release/modm/libmodm.a(croutine.o):/root/project/examples/stm32f746g_discovery/rtos/modm/ext/freertos/inc/freertos/FreeRTOSConfig.h:39: first defined here
ld: build/release/modm/libmodm.a(timers.o):/root/project/examples/stm32f746g_discovery/rtos/modm/ext/freertos/inc/freertos/FreeRTOSConfig.h:39: multiple definition of `modm::platform::fcpu'; build/release/modm/libmodm.a(croutine.o):/root/project/examples/stm32f746g_discovery/rtos/modm/ext/freertos/inc/freertos/FreeRTOSConfig.h:39: first defined here
```
I can't actually find a definition for it outside of the header in the generated code for STM32.

Apart from that, declaring a C++ variable as extern "C" with its mangled name is at least confusing and implementation defined. Maybe we can find a cleaner way to achieve the same result.

--- a/ext/aws/FreeRTOSConfig.h.in
+++ b/ext/aws/FreeRTOSConfig.h.in
@@ -36,12 +36,12 @@ your application. */
 #include <modm/platform/device.hpp>
 
 // declared in modm:platform:clock
-modm_extern_c uint32_t _ZN4modm8platform4fcpuE;
+modm_extern_c uint32_t SystemCoreClock;
 
 #define configUSE_PREEMPTION                    1
 #define configUSE_PORT_OPTIMISED_TASK_SELECTION {{ ("m0" not in core) | int }}
 #define configUSE_TICKLESS_IDLE                 0
-#define configCPU_CLOCK_HZ                      ( _ZN4modm8platform4fcpuE )
+#define configCPU_CLOCK_HZ                      ( SystemCoreClock )
 #define configTICK_RATE_HZ                      ( {{ options["::frequency"] }} )
 #define configMAX_PRIORITIES                    5
 #define configMINIMAL_STACK_SIZE                128

